### PR TITLE
Name lists on invite dialog

### PR DESCRIPTION
**Pull request by luixxiul**
Closes https://github.com/vector-im/element-web/issues/21400
Closes https://github.com/vector-im/element-web/issues/19463 (Closes https://github.com/matrix-org/matrix-react-sdk/pull/8092)

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

Type: Enhancement

after1 after

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Name lists on invite dialog ([\#8046](https://github.com/matrix-org/matrix-react-sdk/pull/8046)). Fixes vector-im/element-web#21400 and vector-im/element-web#19463. Contributed by @luixxiul.

Preview: https://pr8046--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: res/css/views/dialogs/_InviteDialog.scss [hunk 1]
```diff
@@ -404,29 +404,44 @@ limitations under the License.
 
     > div {
         .mx_InviteDialog_multiInviterError_entry {
+            display: grid;
+            grid-template-columns: max-content auto;
```
Comment (luixxiul) [inline res/css/views/dialogs/_InviteDialog.scss:408]: `max-content`: width of the avatar

File: res/css/views/dialogs/_InviteDialog.scss [hunk 2]
```diff
@@ -404,29 +404,44 @@ limitations under the License.
...
+            align-items: center;
+            row-gap: 6px;
+            column-gap: 6px;
```
[the remark this answers is not in the document]
Comment (luixxiul) [reply res/css/views/dialogs/_InviteDialog.scss:411]: I thought these should be left separated as row-gap (between avater + username + domain and the description) and column-gap (between the avatar and username + description) is not necessarily the same value.

**Comment by luixxiul** [comment]
Could I please receive a review from the design team? If something should be changed here, I could happily work on that.

**Comment by luixxiul** [comment]
Could I ask a review by the design team so that I would work on the change that should be done if any?

**Comment by luixxiul** [comment]
@janogarcia Thanks for the input. It looks like at first it needs to detect which dialogs list up users. After that it should not be so difficult to implement the layout.

**Comment by luixxiul** [comment]
Name lists on the invite dialog are going to be styled as below. 

after1 after

Though styles of the room users list can be adjusted to the design input, I think it should be addressed with another PR as it requires intensive testing to avoid regressions which could seriously harm UX.

**Comment by luixxiul** [comment]
@janogarcia could you please review the implementation and let me know what to adjust if any (such as spacing between elements, color, etc.) ?

**Review by janogarcia — COMMENTED** [review]
@luixxiul The Netlify deploy seems to be pending/blocked by GitHub, so I couldn't test your latest commits. From the screenshots it's looking OK (it's reusing the same layout in both places, the layout doesn't fall apart with long strings...).

**Comment by janogarcia** [comment]
The current modals/dialogs could be greatly improved in general, but that would be out of the scope of this issue. 👍

**Comment by luixxiul** [comment]
@janogarcia It seems that the Netlify deployment became available. Would you please try that? Thanks in advance!

**Comment by luixxiul** [comment]
I'm going to fix the conflict soon.

**Review by janogarcia — APPROVED** [review]

**Comment by luixxiul** [comment]
@t3chguy the conflict was fixed.

**Comment by luixxiul** [comment]
If there is something else to be addressed, please let me know.

**Comment by andybalaam** [comment]
The failing test is the current problems with Chrome on Percy - merging.

**Comment by luixxiul** [comment]
@andybalaam thank you!

**Pull request merged**